rustls/src/server/mod.rs

```diff
@@ -143,8 +151,12 @@ impl<'a> ClientHello<'a> {
     /// Get the alpn.
     ///
     /// Returns `None` if the client did not include an ALPN extension
-    pub fn alpn(&self) -> Option<&'a [&'a [u8]]> {
-        self.alpn
+    pub fn alpn(&self) -> Option<impl Iterator<Item = &[u8]>> {
```

Comment:
This is a public API change; this avoids allocating a `Vec` for the protocols and makes it easier to abstract over the differences between the `Acceptor` path and the usual ClientHello handling path.

Reply:
Should this have `Item = &'a [u8]`?

Reply:
Fair enough, fixed that up.